sematic/resolvers/silent_resolver.py

```diff
@@ -16,18 +17,44 @@ class SilentResolver(StateMachineResolver):
     def _schedule_future(self, future: AbstractFuture) -> None:
         self._run_inline(future)
 
+    def _activate_resources(self, future):
+        updated_resources = []
+        for resource in future.props.external_resources:
+            resource = resource.activate(is_local=True)
+            while not (
+                resource.status.state == ResourceState.ACTIVE
+                or resource.status.state.is_terminal()
+            ):
+                resource = resource.update()
+            updated_resources.append(resource)
+        future.props.external_resources = updated_resources
+
+    def _deactivate_resources(self, future):
+        updated_resources = []
+        for resource in future.props.external_resources:
+            resource = resource.deactivate()
```

Comment:
`.activate()`, `.deactivate()`, and `.update()` should be expected to raise, as they involve communication with external resources.

As these exceptions happen on the resources attached to the Futures, I don't believe they could or should be included in the Retry policy (yet, in that feature's current form).

Please catch these exceptions, and start an orderly shutdown of the resolution, including deactivating remaining resources. Please add a unit test for this fatal use case as well.